### PR TITLE
[baselines] Allow PointNav_GPSCompass to support dimensionality 3

### PR DESCRIPTION
**Pull request by Skylion007**
## Motivation and Context

Enables the ResNet baseline to support an integrated pointnav_gps_compass sensor of dimensionality 3. Previously only 2D Integrated PointNav/GPS Compass was supported.

## How Has This Been Tested
* Locally

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [X] All new and existing tests passed.

### CODE REVIEW

File: habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 1]
```diff
@@ -353,14 +353,32 @@ def forward(
...
+                        torch.sin(-goal_observations[:, 1]),
+                    ],
+                    -1,
+                )
+            else:
+                assert (
+                    goal_observations.shape[1] == 3
+                ), "Unsupported dimensionality"
+
+                vertical_angle = torch.sin(goal_observations[:, 2])
```
Comment (mathfac) [inline habitat_baselines/rl/ddppo/policy/resnet_policy.py:371]:
```suggestion
                vertical_angle_sin = torch.sin(goal_observations[:, 2])
```

File: habitat_baselines/rl/ddppo/policy/resnet_policy.py [hunk 2]
```diff
@@ -353,14 +353,32 @@ def forward(
...
+                # Polar Dimensionality 3
```
Comment (mathfac) [inline habitat_baselines/rl/ddppo/policy/resnet_policy.py:372]: That would be great to drop the comment why we multiply horizontal angle sin/cos with vertical sin.

**Review by mathfac — APPROVED** [review]
Thanks!

**Review by erikwijmans — APPROVED** [review]
Probably getting about time to refactor these additional observation encoders.

**Pull request merged**